sims/verilator/Makefile

```diff
@@ -1,112 +1,175 @@
 
-#########################################################################################
-# verilator binary and flags
-#########################################################################################
+#============================================================================
+# selectively enabling printfs
+#============================================================================
+ENABLE_PRINTF_PATTERN ?=
+sim_vsrcs_printf := $(sim_vsrcs:.v=.printf.v)
+
+#============================================================================
+# verilator/cxx binary and flags
+#============================================================================
+# can't use -Wall, firrtl spits out 1000's of warnings that flag as errors
+#VERILATOR := verilator --cc --exe -Wall
 VERILATOR := verilator --cc --exe
 
-CXXFLAGS := \
-	$(CXXFLAGS) -O1 -std=c++11 \
-	-I$(RISCV)/include \
-	-I$(dramsim_dir) \
-	-D__STDC_FORMAT_MACROS \
-	$(EXTRA_SIM_CC_FLAGS)
-
-LDFLAGS := \
-	$(LDFLAGS) \
-	-L$(sim_dir) \
-	-lpthread
-
-VERILATOR_CC_OPTS = \
-	-O3 \
-	-CFLAGS "$(CXXFLAGS) -DTEST_HARNESS=V$(VLOG_MODEL) -DVERILATOR" \
-	-CFLAGS "-I$(build_dir) -include $(build_dir)/$(long_name).plusArgs -include $(build_dir)/verilator.h" \
-	-LDFLAGS "$(LDFLAGS)" \
-	$(RISCV)/lib/libfesvr.a \
-	$(dramsim_lib)
+#----------------------------------------------------------------------------
+# user configs
+#----------------------------------------------------------------------------
+VL8_PROFILE ?= none
+RUNTIME_PROFILING_CFLAGS := $(if $(filter $(VL8_PROFILE),all),-g -pg,)
+RUNTIME_PROFILING_VFLAGS := $(if $(filter $(VL8_PROFILE),all),\
+                              --prof-threads --prof-cfuncs,\
+                              $(if $(filter $(VL8_PROFILE),threads),\
+                                --prof-threads,))
+
+VL8_THREADS ?= 1
+RUNTIME_THREADS := --threads $(VL8_THREADS) --threads-dpi all
+
+VL8_FST_MODE ?= 0
+TRACING_OPTS := $(if $(filter $(VL8_FST_MODE),0),--trace,--trace-fst-thread)
+TRACING_CFLAGS := $(if $(filter $(VL8_FST_MODE),0),,-DCY_FST_TRACE)
+
+#----------------------------------------------------------------------------
+# verilation configuration/optimization
+#----------------------------------------------------------------------------
+VMODE := -O3 --x-assign fast --x-initial fast --noassert \
+          --output-split 10000 --output-split-cfuncs 10000
 
 # default flags added for ariane
 ARIANE_VERILATOR_FLAGS = \
-	--unroll-count 256 \
-	-Werror-PINMISSING \
-	-Werror-IMPLICIT \
-	-Wno-PINCONNECTEMPTY \
-	-Wno-ASSIGNDLY \
-	-Wno-DECLFILENAME \
-	-Wno-UNUSED \
-	-Wno-UNOPTFLAT \
-	-Wno-BLKANDNBLK \
-	-Wno-style \
-	-Wall
+  --unroll-count 256 \
+  -Werror-PINMISSING \
+  -Werror-IMPLICIT \
+  -Wno-PINCONNECTEMPTY \
+  -Wno-ASSIGNDLY \
+  -Wno-DECLFILENAME \
+  -Wno-UNUSED \
+  -Wno-UNOPTFLAT \
+  -Wno-BLKANDNBLK \
+  -Wno-style \
+  -Wall
 
 # normal flags used for chipyard builds (that are incompatible with ariane)
-CHIPYARD_VERILATOR_FLAGS = \
```

Comment:
Why is `--assert` deleted here?

Reply:
because its slower than --noassert. do we use assertions anyways?

Reply:
dramsim uses asserts

Reply:
ok added back